exchanges/bybit/convert.go

```diff
@@ -0,0 +1,154 @@
+package bybit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+var supportedAccountTypes = []WalletAccountType{Funding, Uta, Spot, Contract, Inverse}
+
+var (
+	errUnsupportedAccountType  = errors.New("unsupported account type")
+	errCurrencyCodesEqual      = errors.New("from and to currency codes cannot be equal")
+	errRequestCoinInvalid      = errors.New("request coin must match from coin if provided")
+	errQuoteTransactionIDEmpty = errors.New("quoteTransactionID cannot be empty")
+)
+
+// GetConvertCoinList returns a list of coins you can convert to/from
+func (e *Exchange) GetConvertCoinList(ctx context.Context, accountType WalletAccountType, coin currency.Code, isCoinToBuy bool) ([]CoinResponse, error) {
+	if !slices.Contains(supportedAccountTypes, accountType) {
+		return nil, fmt.Errorf("%w: %q", errUnsupportedAccountType, accountType)
+	}
+
+	params := url.Values{}
+	params.Set("accountType", string(accountType))
+
+	if isCoinToBuy {
+		if coin.IsEmpty() {
+			return nil, currency.ErrCurrencyCodeEmpty
+		}
+		params.Set("coin", coin.Upper().String())
+		params.Set("side", "1")
+	} else {
+		params.Set("side", "0")
+	}
+
+	var resp struct {
+		List []CoinResponse `json:"coins"`
+	}
+
+	return resp.List, e.SendAuthHTTPRequestV5(ctx, exchange.RestSpot, http.MethodGet, "/v5/asset/exchange/query-coin-list", params, nil, &resp, defaultEPL)
+}
+
+// RequestAQuote requests a conversion quote between two coins with the specified parameters.
+func (e *Exchange) RequestAQuote(ctx context.Context, params *RequestAQuoteRequest) (*RequestAQuoteResponse, error) {
```

Comment:
🚧 We don't normally add `A` to methods or structs like this.
🌴 2️⃣ 
```suggestion
// RequestQuote requests a conversion quote between two coins with the specified parameters.
func (e *Exchange) RequestQuote(ctx context.Context, params *RequestQuoteRequest) (*RequestQuoteResponse, error) {
```